poetry/vcs/__init__.py

```diff
@@ -1,22 +1,28 @@
+import os
 import subprocess
-import warnings
 
 from poetry.utils._compat import Path
+from poetry.utils._compat import decode
 
 from .git import Git
 
 
 def get_vcs(directory):  # type: (Path) -> Git
-    directory = directory.resolve()
-
-    for p in [directory] + list(directory.parents):
-        if (p / ".git").is_dir():
-            try:
-                return Git(p)
-            except (subprocess.CalledProcessError, OSError):
-                # Either git could not be found or does not exist
-                warnings.warn(
-                    "git executable could not be found", category=RuntimeWarning
-                )
-
-                return
+    working_dir = Path.cwd()
+    os.chdir(str(directory))
```

Comment:
Shouldn't we keep the `.resolve()` call?

Reply:
Not sure if the is absolutely necessary. But at least it doesn't hurt :smiley: